beacon-chain/sync/initial-sync/blocks_fetcher.go

```diff
@@ -355,8 +454,7 @@ func (f *blocksFetcher) fetchBlocksFromPeer(
 	// peers are dialed first.
 	peers = append(bestPeers, peers...)
 	peers = dedupPeers(peers)
-	for i := 0; i < len(peers); i++ {
-		p := peers[i]
+	for _, p := range peers {
```

Comment:
this isn't a change but this function after requestBlocks maybe should have a check like
```
// Ignore empty responses to avoid rewarding peers that serve nothing.
		if len(blocksResp) == 0 {
			continue
		}
		```

Reply:
why don't we have f.p2p.Peers().Scorers().BlockProviderScorer().Touch(p) at the end before the return robs